src/cron-job.ts

```diff
@@ -0,0 +1,251 @@
+import { ApiObject, CronOptions, Duration, Lazy, Schedule } from 'cdk8s';
+import { Construct } from 'constructs';
+import * as timezone from 'moment-timezone';
+import * as k8s from './imports/k8s';
+import { JobProps } from './job';
+import { RestartPolicy } from './pod';
+import * as workload from './workload';
+
+/**
+ * Concurrency policy for CronJobs.
+ */
+export enum ConcurrencyPolicy {
+  ALLOW = 'Allow',
+  FORBID = 'Forbid',
+  REPLACE = 'Replace',
+}
+
+/**
+ * Properties for `CronJob`.
+ */
+export interface CronJobProps extends workload.WorkloadProps {
+  /**
+   * Properties of the Job that is being scheduled by CronJob.
+   */
+  readonly jobProperties: JobProps;
```

Comment:
Because `JobProps` also extends `WorkloadProps`, this makes it so pod properties can be specified twice. For example:

```ts
new CronJob(this, 'CronJob', {
  jobProperties: {
    containers: ...,
  },
  containers: ...,
});
```

What we need here is actually:

```suggestion
export interface CronJobProps extends JobProps {
```

This would make the API look like:

```ts
new CronJob(this, 'CronJob', {
  containers: ...,
});
```

Which is consistent with what we have for other workload resources. The only potential problem here is with collisions, but since we control the property names, we can always workaround that if needed.

Reply:
Ahh, I see, will update it.